api/v1alpha3/azuremachine_types.go

```diff
@@ -59,6 +59,18 @@ type AzureMachineSpec struct {
 	// AllocatePublicIP allows the ability to create dynamic public ips for machines where this value is true.
 	// +optional
 	AllocatePublicIP bool `json:"allocatePublicIP,omitempty"`
+
+	// SpotVMOptions allows the ability to specify the Machine should use a Spot VM
+	// +optional
+	SpotVMOptions *SpotVMOptions `json:"spotVMOptions,omitempty"`
```

Comment:
There should probably be a conversion in place to backup/restore this value from annotation during conversion to avoid a lossy round trip.